src/uPlot.js

```diff
@@ -1746,12 +1792,12 @@ export default function uPlot(opts, data, then) {
 	}
 
 	function mouseDown(e, src, _x, _y, _w, _h, _i) {
-		if (e == null || filtMouse(e)) {
+		if (src != null || filtMouse(e)) {
 			dragging = true;
 
-			cacheMouse(e, src, _x, _y, _w, _h, _i, true, true);
+			cacheMouse(e, src, _x, _y, _w, _h, _i, true, false);
 
-			if (select.show && (drag.x || drag.y))
+			if (select.show && (dragX || dragY))
```

Comment:
during mousedown, these will always match `drag.x` and `drag.y`, right?

if that's the case, we can leave this line unchanged and actually pull the dynamic `dragX` and `dragY` variables into the local `else` scope (line 1649), since they will no longer be needed anywhere else.